test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,67 @@
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
```

Comment:
For readability and maintainability, I suggest making the test cases independent. i.e. have each of them as a separate instance of the `tests` list. You can have all the non-alerting ones in one test and each alerting in a test of its own. Something like this:

```
tests:
  # Pod is in the Pending state and scheduled, so it will be alerted.
  - interval: 1m
    input_series:

      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Pending"}'
        values: '1x15'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Unknown"}'
        values: '0x15'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Failed"}'
        values: '0x15'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Running"}'
        values: '0x15'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Succeeded"}'
        values: '0x15'
      - series: 'kube_pod_status_unschedulable{namespace="ns-1", pod="pod-1"}'
        values: '0x15'

    alert_rule_test:
      - eval_time: 15m
        alertname: PodsNotReady
        exp_alerts:
          - exp_labels:
              severity: warning
              namespace: ns-1
              pod: pod-1
              phase: Pending
            exp_annotations:
              summary: "Pod pod-1 is not ready"
              description: "Pod pod-1 in namespace ns-1 is not ready for more than 15 minutes."


  # Pod is in the Unknown state and scheduled, so it will be alerted.
  - interval: 1m
    input_series:

      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Pending"}'
        values: '0x15'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Unknown"}'
        values: '1x15'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Failed"}'
        values: '0x15'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Running"}'
        values: '0x15'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Succeeded"}'
        values: '0x15'
      - series: 'kube_pod_status_unschedulable{namespace="ns-1", pod="pod-1"}'
        values: '0x15'

    alert_rule_test:
      - eval_time: 15m
        alertname: PodsNotReady
        exp_alerts:
          - exp_labels:
              severity: warning
              namespace: ns-1
              pod: pod-1
              phase: Pending
            exp_annotations:
              summary: "Pod pod-1 is not ready"
              description: "Pod pod-1 in namespace ns-1 is not ready for more than 15 minutes."

  # Some other test case
  - interval: 1m
    input_series:
    ...
```